addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -3,11 +3,113 @@ odoo.define('mail.messaging.entity.Thread', function (require) {
 
 const { registerNewEntity } = require('mail.messaging.entityCore');
 const { attr, many2many, many2one, one2many, one2one } = require('mail.messaging.EntityField');
+const throttle = require('mail.messaging.utils.throttle');
+const Timer = require('mail.messaging.utils.Timer');
+
+const { _lt } = require('web.core');
 
 function ThreadFactory({ Entity }) {
 
     class Thread extends Entity {
 
+        /**
+         * @override
+         */
+        init(...args) {
+            super.init(...args);
+            /**
+             * Timer of current partner that was currently typing something, but
+             * there is no change on the input for several time. This is used
+             * in order to automatically notify other members that current
+             * partner has stopped typing something, due to making no changes
+             * on the composer for some time.
+             */
+            this._currentPartnerInactiveTypingTimer = new Timer(
+                this.env,
+                () => this.async(() => this._onCurrentPartnerInactiveTypingTimeout()),
```

Comment:
Shouldn't `this.async` be also and/or instead inside `_onCurrentPartnerInactiveTypingTimeout`?

Reply:
Yes, any `await` should be prefixed with `this.async`